apps/root/src/pages/aggregator/swap-container/components/swap/index.tsx

```diff
@@ -777,105 +769,90 @@ const Swap = ({ isLoadingRoute, quotes, fetchOptions, swapOptionsError }: SwapPr
-          done: true,
-          checkForPending: false,
-        };
+      if (signIndex !== -1 && selectedRoute && selectedRoute.tx) {
+        const swapIndex = findIndex(transactions, { type: TRANSACTION_ACTION_SWAP });
 
-        setTransactionsToExecute(newSteps);
-
-        if (
-          newSteps[signIndex + 1] &&
-          newSteps[signIndex + 1].type === TRANSACTION_ACTION_WAIT_FOR_QUOTES_SIMULATION &&
-          selectedRoute &&
-          selectedRoute.tx
-        ) {
-          const swapIndex = findIndex(transactions, { type: TRANSACTION_ACTION_SWAP });
+        if (swapIndex !== -1) {
+          const { signature } = newSteps[swapIndex].extraData as TransactionActionSwapData;
 
-          if (swapIndex !== -1) {
-            const { signature } = newSteps[swapIndex].extraData as TransactionActionSwapData;
-
-            if (signature) {
-              const simulatePromise = simulationService.simulateQuotes(
-                activeWallet.address,
-                quotes,
-                sorting,
-                signature,
-                (isBuyOrder && toValue && to && parseUnits(toValue, to.decimals)) || undefined
-              );
-              return simulatePromise
-                .then((sortedQuotes) => {
-                  if (!sortedQuotes.length) {
-                    handleTransactionSimulationWait(newSteps);
-                    setShouldShowFailedQuotesModal(true);
-                    return null;
+          if (signature) {
+            const simulatePromise = simulationService.simulateQuotes(
+              activeWallet.address,
+              quotes,
+              sorting,
+              signature,
+              (isBuyOrder && toValue && to && parseUnits(toValue, to.decimals)) || undefined
+            );
+            return simulatePromise
+              .then((sortedQuotes) => {
+                if (!sortedQuotes.length) {
+                  handleTransactionSimulationWait(newSteps);
+                  setShouldShowFailedQuotesModal(true);
+                  return null;
+                }
+                const originalQuote = find(sortedQuotes, { swapper: { id: selectedRoute.swapper.id } });
+                const isThereABetterQuote = sortedQuotes[0].swapper.id !== selectedRoute.swapper.id;
+                const isBetteryBy =
+                  isThereABetterQuote &&
+                  parseFloat(
+                    formatCurrencyAmount(
+                      getBetterBy(sortedQuotes[0], selectedRoute, sorting, isBuyOrder) || 0n,
+                      emptyTokenWithDecimals(18),
+                      3,
+                      2
+                    )
+                  ).toFixed(3);
+
+                if (isThereABetterQuote && (Number(isBetteryBy) > 0 || !originalQuote)) {
+                  dispatch(setSelectedRoute(originalQuote || { ...selectedRoute, willFail: true }));
+                  setBetterQuote(sortedQuotes[0]);
+                  setShouldShowBetterQuoteModal(true);
```

Comment:
We should actually remove this, since we are not going to show the modal anymore, we will just update the transaction data that is on top of the steps. We will automatically select a better quote for the user

Reply:
I tried just removing the modal, but it wouldn't apply the quote selection. We have a task created for this, so I think it's better to handle all this changes in BLY-1642

Reply:
cool